File: backend/recpro-backend-lib/src/main/java/de/ubt/ai4/petter/recpro/recpro_backend_lib/recpro/model/recommendation/modeling/userCriteria/environmentalParameterUserCriteria/EnvParamNumericUserCriteria.java
package de.ubt.ai4.petter.recpro.recpro_backend_lib.recpro.model.recommendation.modeling.userCriteria.environmentalParameterUserCriteria;

import jakarta.persistence.DiscriminatorValue;
import jakarta.persistence.Entity;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;
import lombok.experimental.SuperBuilder;

@EqualsAndHashCode(callSuper = true)
@Entity
@Data
@SuperBuilder
@NoArgsConstructor
@AllArgsConstructor

@DiscriminatorValue("NUMERIC")
public class EnvParamNumericUserCriteria extends AbstractEnvParamUserCriteria {

    private double minValue;
    private double maxValue;
    private double value;
}

File: frontend/recpro-frontend-lib/projects/recpro-frontend-lib/src/lib/recommendation/modeling/userCriteria/environmentalParameterUserCriteria/EnvParamNumericUserCriteria.ts
import {AbstractEnvParamUserCriteria} from './AbstractEnvParamUserCriteria';

export class EnvParamNumericUserCriteria extends AbstractEnvParamUserCriteria {
  minValue: number = 0;
  maxValue: number = 0;
  value: number = 0;
}
